### PR TITLE
Added support for AWS Secrets Manager, other providers

### PR DESCRIPTION
**Pull request by kducharm**

### CODE REVIEW

File: bin/secrets_manager [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+// show help and quit
+if (!$command || $getOpt->getOption('help')) {
+    echo $getOpt->getHelpText();
+    exit;
+}
+
+if (isset($options['secrets_manager_class'])) {
+    $project_settings = new Kducharm\ProjectSettings\ProjectSettings();
+    $secrets_class = 'Kducharm\ProjectSettings\\' . $options['secrets_manager_class'];
```
Comment (dangur) [inline bin/secrets_manager:35]: Are there supposed to be 2 backslashes here?

File: bin/secrets_manager [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    $secrets_manager = new $secrets_class($project_settings);
+    //$secrets_provider = new Kducharm\ProjectSettings\SecretsProviders\AwsSecretsMgrSecretsProvider($secrets_manager);
+
+    if (isset($options['secrets_provider_class'])) {
+        /* @var \Kducharm\ProjectSettings\SecretsManager $secrets_manager */
+        $secrets_manager->setSecretProviderClass($options['secrets_provider_class']);
+    }
+
+
+    switch ($command->getHandler()) {
```
Comment (dangur) [inline bin/secrets_manager:51]: Simplify with an `if...else`?

File: bin/secrets_manager [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    $secrets_class = 'Kducharm\ProjectSettings\\' . $options['secrets_manager_class'];
+}
+else {
+    $project_settings = new Kducharm\ProjectSettings\ProjectSettings('environments');
+    $secrets_class = 'Kducharm\ProjectSettings\SampleSecretsManager';
+}
+
+try {
+    $secrets_manager = new $secrets_class($project_settings);
+    //$secrets_provider = new Kducharm\ProjectSettings\SecretsProviders\AwsSecretsMgrSecretsProvider($secrets_manager);
```
Comment (kducharm) [inline bin/secrets_manager]: Removing
```suggestion
    //$secrets_provider = new Kducharm\ProjectSettings\SecretsProviders\AwsSecretsMgrSecretsProvider($secrets_manager);
```

**Review by dangur — CHANGES_REQUESTED** [review]

File: bin/secrets_manager (hunk 1 above)
Comment (kducharm) [reply bin/secrets_manager:35]: Yes otherwise it thinks it's an escaping a single quote

File: bin/secrets_manager (hunk 2 above)
Comment (kducharm) [reply bin/secrets_manager:51]: Figured would add more commands one day, so I can leave it

**Pull request merged**